File: kk-ai/apps/web-admin/src/services/prompt.ts
const API_BASE =
  (import.meta as unknown as { env: Record<string, string> }).env
    .VITE_API_BASE_URL || "http://localhost:8000";

interface ApiError {
  error: string;
  message: string;
  detail?: Record<string, unknown>;
}

async function request<T>(url: string, options: RequestInit = {}): Promise<T> {
  const response = await fetch(`${API_BASE}${url}`, {
    headers: {
      "Content-Type": "application/json",
      ...options.headers,
    },
    ...options,
  });

  if (!response.ok) {
    const error: ApiError = await response.json().catch(() => ({
      error: "UNKNOWN_ERROR",
      message: `HTTP ${response.status}`,
    }));
    throw new Error(error.message || error.error);
  }

  return response.json() as Promise<T>;
}

export interface PromptListItem {
  prompt_id: string;
  name: string;
  category: string;
  version: string;
  description?: string;
}

export interface PromptListResponse {
  items: PromptListItem[];
  total: number;
}

export interface PromptDetail {
  id: string;
  name: string;
  category: string;
  version: string;
  description?: string;
  template: string;
  variables?: Array<{
    name: string;
    default?: string;
    description?: string;
  }>;
}

export interface RegisterPromptRequest {
  id: string;
  name: string;
  category: string;
  template: string;
  variables?: Array<{
    name: string;
    default?: string;
    description?: string;
  }>;
  description?: string;
}

export interface RegisterPromptResponse {
  prompt_id: string;
  version: string;
}

export interface RenderPromptRequest {
  variables: Record<string, string>;
}

export interface RenderPromptResponse {
  prompt_id: string;
  rendered: string;
  variables_used: string[];
  variables_missing: string[];
}

export const promptApi = {
  list(category?: string): Promise<PromptListResponse> {
    const searchParams = new URLSearchParams();
    if (category) searchParams.set("category", category);
    const query = searchParams.toString();
    return request<PromptListResponse>(
      `/v1/prompts${query ? `?${query}` : ""}`,
    );
  },

  get(promptId: string): Promise<PromptDetail> {
    return request<PromptDetail>(`/v1/prompts/${encodeURIComponent(promptId)}`);
  },

  register(data: RegisterPromptRequest): Promise<RegisterPromptResponse> {
    return request<RegisterPromptResponse>("/v1/prompts", {
      method: "POST",
      body: JSON.stringify(data),
    });
  },

  delete(promptId: string): Promise<void> {
    return request<void>(`/v1/prompts/${encodeURIComponent(promptId)}`, {
      method: "DELETE",
    });
  },

  render(
    promptId: string,
    variables: Record<string, string>,
  ): Promise<RenderPromptResponse> {
    return request<RenderPromptResponse>(
      `/v1/prompts/${encodeURIComponent(promptId)}/render`,
      {
        method: "POST",
        body: JSON.stringify({ variables }),
      },
    );
  },
};

File: kk-ai/services/service-prompt/app/models/prompt.py
"""Pydantic models for Prompt Center."""

from typing import Any

from pydantic import BaseModel, Field


class PromptVariable(BaseModel):
    """A variable definition in a prompt template."""

    name: str
    required: bool = False
    description: str | None = None
    default: Any | None = None
    type: str = "string"
    example: str | None = None


class PromptTemplate(BaseModel):
    """A prompt template definition."""

    id: str
    name: str
    category: str = "uncategorized"
    version: str = "1.0.0"
    description: str | None = None
    author: str | None = None
    tags: list[str] = Field(default_factory=list)
    variables: list[PromptVariable] = Field(default_factory=list)
    template: str = ""
    messages: list[dict[str, str]] | None = None


class PromptListItem(BaseModel):
    """A single item in prompt list."""

    prompt_id: str
    name: str
    category: str
    version: str
    description: str | None = None


class PromptListResponse(BaseModel):
    """Response for prompt list."""

    items: list[PromptListItem]
    total: int


class RenderPromptRequest(BaseModel):
    """Request to render a prompt."""

    variables: dict[str, Any] = Field(default_factory=dict)


class RenderPromptResponse(BaseModel):
    """Response from prompt rendering."""

    prompt_id: str
    rendered: str
    variables_used: list[str]
    variables_missing: list[str] = Field(default_factory=list)


class RegisterPromptRequest(BaseModel):
    """Request to register a new prompt."""

    id: str = Field(..., min_length=1)
    name: str = Field(..., min_length=1)
    category: str = "uncategorized"
    description: str | None = None
    variables: list[PromptVariable] = Field(default_factory=list)
    template: str = ""


class RegisterPromptResponse(BaseModel):
    """Response after registering a prompt."""

    prompt_id: str
    version: str = "1.0.0"
    status: str = "registered"


class MCPPromptArgument(BaseModel):
    """MCP-style prompt argument."""

    name: str
    description: str | None = None
    required: bool = False


class MCPPromptDetail(BaseModel):
    """MCP-style prompt detail."""

    name: str
    description: str | None = None
    arguments: list[MCPPromptArgument] = Field(default_factory=list)
